src/main/ruby/core/enumerable.rb

```diff
@@ -471,17 +471,27 @@ def inject(initial=undefined, sym=undefined, &block)
   end
   alias_method :reduce, :inject
 
-  def all?
-    if block_given?
+  def all?(pattern = undefined)
+    if !undefined.equal?(pattern)
+      each do |*e|
+        e = e.first unless e.size > 1
```

Comment:
Can you use `Truffle.single_block_arg`? (see below)
It deals with this pattern of extracting the argument if there is only 1 argument.

Reply:
I didn't understand what `Truffle.single_block_arg` was, but that's much nicer. Thanks!

Reply:
I think it's originally a Rubinius thing.